### PR TITLE
Featue/pagination

### PR DESCRIPTION
**Pull request by nogahaskal**
Create server file pagination for shared files. 
Make users call more efficient using dictionary.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -91,7 +91,7 @@ services:
     ports:
       - "6379:6379"
   permission-service:
-    image: "drivehub.azurecr.io/meateam/permission-service:v2.0.0"
```
Comment (Shahar-Y) [inline docker-compose.yml:94]: change this

File: src/api/files.js [hunk 2]
```diff
@@ -37,16 +37,16 @@ export async function getFoldersByFolder(parent) {
 /**
  * fetchSharedFiles fetch all the shared files in the current folder
  */
-export async function fetchSharedFiles() {
-  const res = await Axios.get(`${baseURL}/api/files?shares&appId=drive`);
-  const files = res.data;
-  return files;
+export async function fetchSharedFiles(pageNum) {
+  const res = await Axios.get(`${baseURL}/api/files?shares&appId=drive&pageSize=10&pageNum=${pageNum}`);
```
Comment (Shahar-Y) [inline src/api/files.js:41]: change pageSize to const

File: src/api/files.js [hunk 3]
```diff
@@ -37,16 +37,16 @@ export async function getFoldersByFolder(parent) {
...
+  const permissions = res.data;
+  return permissions;
 }
 
-export async function fetchExternalTransferdFiles() {
-  const res = await Axios.get(`${baseURL}/api/files?shares&appId=dropbox`);
-  const files = res.data;
-  return files;
+export async function fetchExternalTransferdFiles(pageNum) {
+  const res = await Axios.get(`${baseURL}/api/files?shares&appId=dropbox&pageSize=10&pageNum=${pageNum}`);
```
Comment (Shahar-Y) [inline src/api/files.js:47]: change pageSize to const

File: src/components/files/list/FileTable.vue [hunk 4]
```diff
@@ -49,45 +48,38 @@
...
   computed: {
     ...mapGetters(["chosenFiles"]),
   },
   data() {
     return {
-      selected: [],
+      selected: this.chosenFiles,
       page: 1,
-      itemsPerPage: 7,
+      itemsPerPage: 10,
```
Comment (Shahar-Y) [inline src/components/files/list/FileTable.vue:82]: should also be the same const from the config

File: src/store/modules/files.js [hunk 5]
```diff
@@ -97,18 +92,14 @@ const actions = {
...
         const isOwner = isFileOwner(file.ownerId);
-        file.owner = isOwner ? "אני" : "???";
-        commit("addFile", file);
 
-        if (!isOwner) {
-          const formattedFile = file;
-          formattedFile.owner = await getFileOwnerName(file.ownerId);
-          commit("updateFile", formattedFile);
-        }
+        formattedFile.owner = isOwner ? "אני" : await getFileOwnerName(file.ownerId);
```
Comment (Shahar-Y) [inline src/store/modules/files.js:101]: I'd prefer there won't be hebrew in the code

File: src/store/modules/search.js [hunk 6]
```diff
@@ -4,27 +4,19 @@ import { isFileOwner, getFileOwnerName, getExternalFileOwnerName } from "@/utils
...
-          commit("addFile", file);
-
-          if (!isOwner) {
-            const formattedFile = file;
-            formattedFile.owner = await getFileOwnerName(file.ownerId);
-            commit('updateFile', formattedFile);
-          }
+          const formattedFile = file;
+          const isOwner = isFileOwner(file.ownerId);
+          formattedFile.owner = isOwner ? "אני" : await getFileOwnerName(file.ownerId);
```
Comment (Shahar-Y) [inline src/store/modules/search.js:18]: hebrew

**Review by Shahar-Y — CHANGES_REQUESTED** [review]

**Comment by Shahar-Y** [comment]
Also, update the CHANGELOG.md

**Comment by Shahar-Y** [comment]
@nogahaskal please refer to the cr

File: src/api/files.js (hunk 2 above)
Comment (nogahaskal) [reply src/api/files.js:41]: done.

File: docker-compose.yml (hunk 1 above)
Comment (nogahaskal) [reply docker-compose.yml:94]: done.

File: src/api/files.js (hunk 3 above)
Comment (nogahaskal) [reply src/api/files.js:47]: done.

File: src/components/files/list/FileTable.vue (hunk 4 above)
Comment (nogahaskal) [reply src/components/files/list/FileTable.vue:81]: done.

File: src/store/modules/files.js (hunk 5 above)
Comment (nogahaskal) [reply src/store/modules/files.js:101]: I have problem to use in this file the i18n because it's a simple js that doesn't include the vue interface.

**Review by Shahar-Y — APPROVED** [review]

**Pull request merged**